packages/fast-components-react-base/src/toolbar/toolbar.props.ts

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import {
+    ManagedClasses,
+    ToolbarClassNameContract,
+} from "@microsoft/fast-components-class-name-contracts-base";
+import { Orientation } from "@microsoft/fast-web-utilities";
+
+export interface ToolbarManagedClasses extends ManagedClasses<ToolbarClassNameContract> {}
+export interface ToolbarUnhandledProps extends React.HTMLAttributes<HTMLDivElement> {}
+export interface ToolbarHandledProps extends ToolbarManagedClasses {
+    /**
+     * The toolbar children
+     */
+    children?: React.ReactNode;
+
+    /**
+     * When true, the toolbar will immediately place focus on the appropriate focusable child when the toolbar mounts
+     */
+    enableAutoFocus?: boolean;
+
+    /**
+     * The index of the child which gets focused when the toolbar initially gets focus
+     */
+    initialFocusIndex?: number;
+
+    /**
+     * Whether the toolbar is horizontal or vertical.  Default is horizontal.
+     * Note that actual layout of the toolbar is controlled via css styling and
+     * this prop mostly affects whether navigation is controlled by the up/down arrows
+     * or the left/right arrows
+     */
+    orientation?: Orientation;
+
+    /**
+     * The toolbar component will only pass focus to children who have been assigned valid aria roles -
+     * https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques
+     * The default list isToolbar.DefaultFocusableRoles and contains most "widget" types.
+     * Authors can pass their own list of valid roles instead.
+     */
+    focusableRoles?: string[];
+
+    /**
+     * Whether child items are focusable when disabled.
+     * Detecting disabled elements is based onelements having a "aria-disabled" tag set to true when disabled
```

Comment:
If I put a button into the toolbar, do I need to set `aria-disabled` on that instead of `disabled`?

Reply:
That's what we are detecting  Context menu detects disabled children in this way.  Can/should we try to detect the disabled attribute as well?

Reply:
Context Menu has a fixed set of things that can go in it, which is why I believe we are looking for the role. There are not semantic elements for `menuitem`, `menuitemcheckbox`, `menuitemradio` so the role would have to be present. This scenario requires a much more broad application.

Reply:
Yes, it does.  But we do need to be able to differentiate between focusable and non-focusable elements in the toolbar.  Another approach could be to have authors specifically label or declare those elements in props somehow.

Reply:
...but this is about detecting disabled state.

Reply:
Sure - I just think we likely need to consider other options here. Requiring a specific implementation of aria, which is not required normally, goes against web standards and best practices for implementors. IE, adding a role of button on a button or adding an attribute of aria-disabled on an input or button which has a supported attribute of "disabled"

Reply:
improved disabled state detection to include "disabled" attribute